### PR TITLE
feat(patterns/textarea): update textarea pattern to use it in forms

### PR DESCRIPTION
**Pull request by tiloyi**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

**Update textarea pattern to use it in forms**

GitHub issue number or Jira issue URL: [ID-676](https://adeo.kanbanize.com/ctrl_board/10/cards/676/details/)

### CODE REVIEW

**Comment by tiloyi** [comment]
Deployment url: https://featid676formtextarea-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com

File: src/docs/Components/Form/TextArea/code.mdx [hunk 1]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
 @import 'components/_c.textarea';
+
+@import 'components/_c.fields';
+```
+
+> Note that the import order is important to get the right rendering of the component.
+
+### Usage
+
+To use the textarea in your form, you must use the following 3 CSS classes:
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:27]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 2]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
 
-## Basic usage
+<Preview path="default" />
+
+### Behaviors
 
-Apply the classes `mc-textarea` on a standard `textarea` html tag :
+#### Help text
+
+You have the possibility to define a help text when using your textarea.
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:55]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 3]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
+  - the span `mc-field__help` must always have an `id`
+  - the `aria-describedby` attribute must be added on the textarea element and must refer to the id of the help text
+
+</Highlight>
+
+<Preview path="help-text" />
+
+#### Requirement
+
+When the textarea is mandatory you must specify it to the user.
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:80]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 4]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
+  Label
+  <span class="mc-field__requirement">
+    mandatory
+  </span>
+</label>
+```
+
+<Highlight theme="tips" title="Semantic rule">
+
+  When the textarea is mandatory the `required` attribute must be added on the textarea element
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:95]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 5]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
+
+</Highlight>
+
+<Preview path="requirement" />
+
+### States and Validation
+
+#### Valid
+
+To indicate that a field is valid, the `is-valid` class must be added to the textarea element.
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:105]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 6]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
+To indicate that a field is not valid several steps are to be implemented:
+
+- Add the `is-invalid` class and the `aria-invalid` attribute to the textarea element
+- You can add in addition a text detailing the error. This text must be added in a **span** with the class `mc-field__error-message` and positioned below the field
+
+<Highlight theme="tips" title="Accessibility rules">
+
+  If you use an error message you must comply with the following accessibility rules:
+  - the span `mc-field__error-message` must always have an `id`
+  - the `aria-describedby` attribute must be added on the textarea element and must refer to the id of the error message
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:120]: text area

File: src/docs/Components/Form/TextArea/code.mdx [hunk 7]
```diff
@@ -3,39 +3,161 @@ title: 'Code'
...
+
+</Highlight>
+
+<Preview path="invalid" />
+
+#### Disabled
+
+To indicate that textarea is disabled, the `disabled` attribute must be added to the textarea element.
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/code.mdx:128]: text area

File: src/docs/Components/Form/TextArea/index.mdx [hunk 8]
```diff
@@ -9,7 +9,7 @@ status:
 
 > A text area is used to fill multi-line text. They share appearence and states with [text inputs](/Components/TextInput/).
 
-<Preview path="TextArea" nude />
+<Preview path="textarea" nude />
 
 Text areas are used for content **with multiple sentence like** :
```
Comment (nherwegh) [inline src/docs/Components/Form/TextArea/index.mdx:14]: no space before ":"

**Pull request merged**